### PR TITLE
doc: add `make html-live` and `html-live-fast`

### PR DESCRIPTION
**Pull request by JordanYates**
Add two new commands that automatically locally host the generated documentation upon completion and rebuild/rehost when the input files change without any user interaction.

For more info see: https://pypi.org/project/sphinx-autobuild/

### CODE REVIEW

**Review by gmarull — COMMENTED** [review]
does this work for rst files out of doc/ folder, eg board/samples docs?

**Comment by JordanYates** [comment]
> does this work for rst files out of doc/ folder, eg board/samples docs?

It doesn't automatically pick them up, but they can be added with additional `--watch` parameters

**Review by aescolar — COMMENTED** [review]
No complaints from me, but @gmarull and @kartben, and probably @carlescufi should  approve.
Thanks for remembering about the release notes :)

**Comment by JordanYates** [comment]
> > > does this work for rst files out of doc/ folder, eg board/samples docs?
> > 
> > 
> > It doesn't automatically pick them up, but they can be added with additional `--watch` parameters
> 
> can you try?

`--watch` only takes folders, not patterns, so all I could really do is add `zephyr/boards` and `zephyr/samples` to the watch list. While that would work, it triggers a rebuild no matter which files are changes (`.dtsi`, `.yaml`, etc).

**Comment by kartben** [comment]
> > > > does this work for rst files out of doc/ folder, eg board/samples docs?
> > > 
> > > 
> > > It doesn't automatically pick them up, but they can be added with additional `--watch` parameters
> > 
> > 
> > can you try?
> 
> `--watch` only takes folders, not patterns, so all I could really do is add `zephyr/boards` and `zephyr/samples` to the watch list. While that would work, it triggers a rebuild no matter which files are changes (`.dtsi`, `.yaml`, etc).

It might be possible to use `--ignore` or `re-ignore` to exclude some file patterns but it looks as if it would be better if there was a pattern-based way to clearly indicate the things to watch in the first place.

**Review by kartben — DISMISSED** [review]

**Review by fabiobaltieri — DISMISSED** [review]

**Comment by JordanYates** [comment]
Reparsing the argument list in `conf.py` no longer works properly when being run through `sphinx-autobuild`, updated to get the one variable of interest from the environment instead.

**Comment by carlescufi** [comment]
> Reparsing the argument list in `conf.py` no longer works properly when being run through `sphinx-autobuild`, updated to get the one variable of interest from the environment instead.

Seems the doc build is failing @JordanYates

**Comment by JordanYates** [comment]
> > Reparsing the argument list in `conf.py` no longer works properly when being run through `sphinx-autobuild`, updated to get the one variable of interest from the environment instead.
> 
> Seems the doc build is failing @JordanYates

Should be resolved, `ZEPHYR_BUILD` was the html dir, not the root build dir.

**Review by fabiobaltieri — DISMISSED** [review]

**Review by kartben — APPROVED** [review]

**Comment by JordanYates** [comment]
Rebased on `v3.7.0`

File: doc/releases/release-notes-3.7.rst [hunk 1]
```diff
@@ -556,6 +556,13 @@ Build system and Infrastructure
   * Twister now supports ``--flash-before`` parameter that allows flashing DUT before
     opening serial port (:github:`47037`).
 
+Documentation
+*************
+
+ * Added two new build commands, ``make html-live`` and ``make html-live-fast``, that automatically locally
+   host the generated documentation. They also automatically rebuild and rehost the documentation when changes
+   to the input ``.rst`` files are detected on the filesystem.
```
Comment (kartben) [inline doc/releases/release-notes-3.7.rst:564]: needs to move to new release notes :)

**Review by kartben — CHANGES_REQUESTED** [review]

**Review by kartben — APPROVED** [review]

**Review by gmarull — APPROVED** [review]

**Pull request merged**